nixos/modules/services/networking/globalprotect-vpn.nix

```
@@ -0,0 +1,43 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.globalprotect;
+
+  execStart = if cfg.csdWrapper == null then
+      "${pkgs.globalprotect-openconnect}/bin/gpservice"
+    else
+      "${pkgs.globalprotect-openconnect}/bin/gpservice --csd-wrapper=${cfg.csdWrapper}";
+in
+
+{
+  options.services.globalprotect = {
+    enable = mkEnableOption "globalprotect";
+
+    csdWrapper = mkOption {
+      description = ''
+        A script that will produce a Host Integrity Protection (HIP) report,
+        as described at https://www.infradead.org/openconnect/hip.html
+      '';
+      default = null;
+      example = "${pkgs.openconnect}/libexec/openconnect/hipreport.sh";
+      type = types.nullOr types.path;
+    };
+  };
+
+  config = {
+    services.dbus.packages = [ pkgs.globalprotect-openconnect ];
+
+    systemd.services.gpservice = {
+      description = "GlobalProtect openconnect DBus service";
+      serviceConfig = {
+        Environment="\"LANG=en_US.utf8\"";
```

Also is this really necessary? `LANG` is already added to the environment of systemd services by default as far as I can see (`systemctl show-environment`). If necessary it probably should use `config.i18n.defaultLocale`.

```suggestion
      description = "GlobalProtect openconnect DBus service";
      environment = {
        LANG = "en_US.utf8";
      };
      serviceConfig = {
```

---

I'm not really sure if it's necessary.  I just transcribed https://github.com/yuezk/GlobalProtect-openconnect/blob/master/GPService/systemd/gpservice.service.  I can test at work tomorrow without it, if it's a concern.

---

Test was quicker than I thought it would be; it works without it, so I've removed it.